data/sqlutil/macros.go

```diff
@@ -124,110 +124,73 @@ var DefaultMacros = Macros{
 	"column":     macroColumn,
 }
 
-type Macro struct {
-	Name string
-	Args []string
+type macroMatch struct {
+	full string
+	args []string
 }
 
 // getMacroMatches extracts macro strings with their respective arguments from the sql input given
 // It manually parses the string to find the closing parenthesis of the macro (because regex has no memory)
-func getMacroMatches(input string, name string) ([]Macro, error) {
-	macroName := fmt.Sprintf("\\$__%s\\b", name)
-	matchedMacros := []Macro{}
-	rgx, err := regexp.Compile(macroName)
+func getMacroMatches(input string, name string) ([]macroMatch, error) {
+	rgx, err := regexp.Compile(fmt.Sprintf(`\$__%s\b`, name))
 
 	if err != nil {
 		return nil, err
 	}
 
-	// get all matching macro instances
-	matched := rgx.FindAllStringIndex(input, -1)
+	var matches []macroMatch
+	for _, window := range rgx.FindAllStringIndex(input, -1) {
+		start, end := window[0], window[1]
+		args, length := parseArgs(input[end:])
+		if length < 0 {
+			return nil, fmt.Errorf("failed to parse macro arguments (missing close bracket?)")
+		}
+		matches = append(matches, macroMatch{full: input[start : end+length], args: args})
+	}
+	return matches, nil
+}
 
-	if matched == nil {
-		return nil, nil
+// parseArgs looks for a bracketed argument list at the beginning of argString.
+// If one is present, returns a list of whitespace-trimmed arguments and the
+// length of the string comprising the bracketed argument list.
+func parseArgs(argString string) ([]string, int) {
+	if !strings.HasPrefix(argString, "(") {
+		return nil, 0 // single empty arg for backwards compatibility
 	}
 
-	for matchedIndex := 0; matchedIndex < len(matched); matchedIndex++ {
-		var macroEnd = 0
-		var argStart = 0
-		// quick exit from the loop, when we encounter a closing bracket before an opening one (ie "($__macro)", where we can skip the closing one from the result)
-		var forceBreak = false
-		macroStart := matched[matchedIndex][0]
-		inputCopy := input[macroStart:]
-		cache := make([]rune, 0)
-
-		// find the opening and closing arguments brackets
-		for idx, r := range inputCopy {
-			if len(cache) == 0 && macroEnd > 0 || forceBreak {
-				break
+	var args []string
+	depth := 0
+	arg := []rune{}
+
+	for i, r := range argString {
+		switch r {
+		case '(':
+			depth++
+			if depth == 1 {
+				// don't include the outer bracket in the arg
+				continue
+			}
+		case ')':
+			depth--
+			if depth == 0 {
+				// closing bracket
+				args = append(args, strings.TrimSpace(string(arg)))
+				return args, i + 1
 			}
-			switch r {
-			case '(':
-				cache = append(cache, r)
-				if argStart == 0 {
-					argStart = idx + 1
-				}
-			case ' ':
-				// when we are inside an argument, we do not want to exit on space
-				if argStart != 0 {
-					continue
-				}
-				fallthrough
```

Comment:
This case was added to fix https://github.com/grafana/sqlds/issues/82, but left a more general issue (https://github.com/grafana/grafana-plugin-sdk-go/issues/859) unaddressed. Both are now handled by the `strings.HasPrefix` check at the top of the new `parseArgs`.